src/components/Checkout/CheckoutDialog.tsx

```diff
@@ -101,31 +108,82 @@ export const CheckoutDialog: React.FC<CheckoutDialogProps> = ({
       );
       let data = null;
       if (isWelcomeBasket) {
-        data = await processWelcomeBasket(user, loggedInUserId, allItems, residentInfo);
+        data = await processWelcomeBasket(
+          user,
+          loggedInUserId,
+          allItems,
+          residentInfo,
+        );
       } else {
-        data = await processGeneralItems(user, loggedInUserId, allItems, residentInfo);
+        data = await processGeneralItems(
+          user,
+          loggedInUserId,
+          allItems,
+          residentInfo,
+        );
       }
 
       if (data.error) {
-        throw new Error(
-          `status: ${data.error.status}, message: ${data.error.message}`,
-        );
+        const errorMessage =
+          data.error.message || 'An unexpected error occurred during checkout';
+        throw new Error(errorMessage);
       }
 
       const result = data.value[0];
       if (result.Status === 'Success') {
         setActiveSection('');
-        setResidentInfo({id: 0, name: '', unit: {id: 0, unit_number: ''}, building: { id: 0, code: '', name: '' }});
+        setResidentInfo({
+          id: 0,
+          name: '',
+          unit: { id: 0, unit_number: '' },
+          building: { id: 0, code: '', name: '' },
+        });
         fetchData();
         setStatusMessage('Transaction Successful');
         onClose();
         onSuccess();
       } else {
-        throw new Error(result.message);
+        const errorMessage =
+          result.message || 'Checkout failed - please try again';
+        throw new Error(errorMessage);
       }
     } catch (error) {
       console.error('Transaction failed:', error);
-      setStatusMessage('Transaction failed: ' + error);
+
+      // Create a user-friendly error message
```

Comment:
I would remove the comment. It doesn't clarify anything in the code. The code is self describing.